src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -919,6 +925,11 @@ static ucs_status_t uct_tcp_sockcm_ep_server_create(uct_tcp_sockcm_ep_t *tcp_ep,
 
     UCS_ASYNC_BLOCK(async);
 
+    if (tcp_ep->state & UCT_TCP_SOCKCM_EP_FAILED) {
```

Comment:
Should we move this check to line 913, before the 'if' in line 917? or even to the beginning of this function.
because if the tcp_ep's state is FAILED, there is no point in attempting the handler removal in line 918. 
wdyt?

Reply:
moved right after for params check, because we have to check `FAILED` flag when async is blocked